### ISSUE TITLE
Converting division to PTX fails

### ISSUE THREAD

**Issue by atzannes** [ufora/ufora#242]
```self.compareCudaToCPU("fun((a,b)) { a/b }", "(2, 1)")```

Fails with:
```
AssertionError: Failed with fun((a,b)) { a/b } on (2, 1): Exception(
    (
        "Code returns multiple types.", 
        (BA00CC2F65A2..., ((f: <eval>.<unnamed function>.f(...*), i: (2, 1)),))), 
    nothing)
```

**Comment by braxtonmckee** [comment]
Yeah this is because of the exception path. I think the right thing to do
is to modify it so that exceptions cause that particular frame to exit
early. We should keep a mask of which threads exited with real values. We
can then execute the early-exit threads on the CPU to capture whatever
actually happened. In the short-run, let's not worry about this...

On Mon, Mar 21, 2016 at 5:55 PM, Alexandros Tzannes <
notifications@github.com> wrote:

> self.compareCudaToCPU("fun((a,b)) { a/b }", "(2, 1)")
>
> Fails with:
>
> AssertionError: Failed with fun((a,b)) { a/b } on (2, 1): Exception(
>     (
>         "Code returns multiple types.",
>         (BA00CC2F65A2..., ((f: <eval>.<unnamed function>.f(...*), i: (2, 1)),))),
>     nothing)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ufora/ufora/issues/242>
>



-- 
-------------------------------------------------------------------------------------------
Braxton McKee                             CEO / Ufora
www.twitter.com/braxtonmckee       www.ufora.com
